hs/src/Reach/Verify/SMTAst.hs

```diff
@@ -170,26 +175,43 @@ instance Pretty SMTTrace where
 
 instance PrettySubst SMTTrace where
   prettySubst (SMTTrace lets tk dv) = do
-    lets' <- mapM prettySubst lets
+    let (inlinable, others) = partition isWitness lets
+    inlinable' <- mapM (prettySubst . toWitness) inlinable
+    lets' <- mapM prettySubst others
     return $
+      "  // Violation Witness" <> hardline <> hardline <>
+      concatWith (surround hardline) inlinable' <> hardline <> hardline <>
+      "  // Theorem Formalization" <> hardline <> hardline <>
       concatWith (surround hardline) lets' <> hardline <>
       "  " <> pretty tk <> parens (pretty dv) <> ";" <> hardline
+    where
+      -- xxx nail down how to get witness variables
```

Comment:
Why can't the annotation on `SMTLet` be trusted? It may be useful for the PR to have some updates to a few choice test outputs so I can follow what the current thing prints out with